livepeer_cli: use the O's currently registered Service URI as default address

emranemran opened the pull request:
When changing config options for the Orchestrator via livepeer_cli, the
public host:port of the node is requested for the configuration to be
applied successfully. This host:port address defaults to using the
ip-address of the node instead of using the global domain name that is
registered when setting up an O node. Use the registered Service URI as
the default option such that the entry can be left blank when
configuring options via the livepeer-cli.

**What does this pull request do? Explain your changes. (required)**
(see above commit desc)

**Specific updates (required)**
(see above commit desc)

**How did you test each of these updates (required)**
* testing locally using both off-chain and on-chain OT nodes. 

**Does this pull request close any open issues?**
fix \#2293

**Checklist:**
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] Read the [contribution guide](./doc/contributing.md)
- [X] `make` runs successfully
- ~~[ ] All tests in `./test.sh` pass~~
- ~~[ ] README and other documentation updated~~
- [x] [Pending changelog](./CHANGELOG_PENDING.md) updated

emranemran commented:
Tested locally with both off-chain and on-chain BOT nodes - it's working as expected and addresses the ticket's need.

leszko commented on cmd/livepeer_cli/wizard_transcoder.go:
```diff
@@ -66,13 +67,16 @@ func (w *wizard) promptOrchestratorConfig() (float64, float64, int, int, string)
 	pricePerUnit := w.readDefaultInt(0)
 
-	addr := myHostPort()
+	if w.offchain {
```
I think that this `if` does not make sense, because you can never run this function in the offchain mode. Orchestrator activation and setting orchestrator config is only for on-chain.

However, I think you need another condition here.
1. If `orch.ServiceURI` is already set (orchestrator is already registered on-chain), then use `orch.ServiceURI`. 
2. Otherwise, use `myHostPort()`

emranemran replied:
Got it, that makes more sense. I was thinking there might be some use cases for off-chain where `myHostPort` needs  is used as-is. Will update PR.

emranemran replied:
Addressed all comments. Also added a condition to not allow orchestrator config settings to be changed if off-chain is detected based on your comments. 

Only outstanding issue was that test.sh fails locally for unrelated reasons - likely related to clang compiler that gets used by default on my Mac M1. The tests executed in this PR all pass so this should be OK to merge. If it's alright, I can follow up with a fix for test.sh failures on Macs as part of a separate ticket. 
```
/opt/homebrew/Cellar/ffmpeg/5.0.1/include/libavformat/avformat.h:2161:41: note: passing argument to parameter 'decoder_ret' here
extras.c:193:13: error: implicit declaration of function 'avfilter_compare_sign_bypath' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
extras.c:193:13: note: did you mean 'lpms_compare_sign_bypath'?
extras.c:191:5: note: 'lpms_compare_sign_bypath' declared here
extras.c:204:13: error: implicit declaration of function 'avfilter_compare_sign_bybuff' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
extras.c:204:13: note: did you mean 'lpms_compare_sign_bybuffer'?
extras.c:202:5: note: 'lpms_compare_sign_bybuffer' declared here
```

leszko replied:
If tests pass on GH Actions, that's fine. M1 may fail for some unrelated reasons. I also experience some test failures on M1.

leszko reviewed (changes requested):
Thanks for the PR @emranemran Added one comment. 

Other than that, could you improve the PR description with the following points?
1. Change `#2293` to `fix #2293`. Then, merging this PR will automatically close the related GH Issue
2. Update `PENDING_CHANGELOG.md`
3. Fill the checklist:
  - Run `./test` locally and check it if it works
  - Cross out `README and other documentation updated` if there is README update related to this PR
  - Update `PENDING_CHANGELOG.md` and check `Pending changelog updated`

emranemran committed:
livepeer_cli: use O's currently registered ServiceURI as default address

When changing config options for the Orchestrator via livepeer_cli, the
public host:port of the node is requested for the configuration to be
applied successfully. This host:port address defaults to using the
ip-address of the node instead of using the global domain name that is
registered when setting up an O node. Use the registered Service URI as
the default option such that the entry can be left blank when
configuring options via the livepeer-cli.

Update CHANGELOG_PENDING.md

Co-authored-by: Rafał Leszko <rafal@livepeer.org>

leszko commented on CHANGELOG_PENDING.md:
```diff
@@ -20,6 +20,7 @@
 ### Bug Fixes 🐞
 
 #### CLI
+- \#2416 Use the O's currently registered Service URI as default address
```
```suggestion
- \#2416 Use the O's currently registered Service URI as default address (@emranemran)
```

leszko reviewed (approved):
Added one more minor suggestion, other than that, LGTM 👍
